lib/compiler/passes/report-incorrect-plucking.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+const GrammarError = require("../../grammar-error");
+const visitor = require("../visitor");
+
+//
+// Compiler pass to ensure the following are enforced:
+//
+//   - plucking can not be done with an action block
+//   - cannot pluck a semantic predicate
+//
+function reportIncorrectPlucking(ast) {
+  const check = visitor.build({
+    action(node) {
+      check(node.expression, true);
+    },
+
+    labeled(node, action) {
+      if (node.pick) {
+        if (action) {
+          throw new GrammarError(
+            "\"@\" cannot be used with an action block "
+              + "at line " + node.location.start.line + ", "
+              + "column " + node.location.start.column + ".",
+            node.location
+          );
+        }
+
+        if (node.expression.type.startsWith("semantic_")) {
```

Comment:
It is better to forbid that on the grammar level. I understand that you just picked up the existing implementation, so that can be done in the follow up PR, but personally I prefer to keep the commit history as clean as we can (without "fix" commits). So if it possible it is better to make now

Reply:
The easiest fix is to just call `error` inside of LabeledExpression if the type of the expression is bad, but that feels like a cheat.  It's important that `@$[a-z]i+` works, which is also in PrefixedExpression.  Refactoring the grammar to get all of that nuance in feels like it would be more confusing than the quick fix, though.

Opinions?